packages/atomic/src/components/commerce/atomic-commerce-recommendation-list/atomic-commerce-recommendation-list.ts

```diff
@@ -0,0 +1,530 @@
+   */
+  @property({reflect: true, type: String}) public density: ItemDisplayDensity =
+    'normal';
+  /**
+   * The expected size of the image displayed on the recommended products.
+   */
+  @property({reflect: true, attribute: 'image-size', type: String})
+  public imageSize: ItemDisplayImageSize = 'small';
+
+  /**
+   * The number of products to display per page.
+   * The products will be displayed in a carousel if this property is set.
+   * This does not affect the display of the list itself, only the number of recommendation pages.
+   * If you want to display the recommendations in a carousel with a single row, set the `--atomic-recs-number-fof-columns` CSS variable to the same value as this property.
+   */
+  @property({reflect: true, attribute: 'products-per-page', type: Number})
+  public productsPerPage?: number;
+
+  /**
+   * The [heading level](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements) to use for the heading label, from 1 to 6.
+   */
+  @property({reflect: true, attribute: 'heading-level', type: Number})
+  public headingLevel = 0;
+
+  @watch('productsPerPage')
+  public async watchNumberOfRecommendationsPerPage() {
+    this.currentPage = 0;
+  }
+
+  /**
+   * Sets a rendering function to bypass the standard HTML template mechanism when rendering products.
+   * You can use this function while working with web frameworks that don't use plain HTML syntax, e.g., React, Angular, or Vue.
+   *
+   * Do not use this method if you integrate Atomic in a plain HTML implementation.
+   *
+   * @param productRenderingFunction
+   */
+
+  public async setRenderFunction(
+    productRenderingFunction: ItemRenderingFunction
+  ) {
+    this.itemRenderingFunction = productRenderingFunction;
+  }
+
+  /**
+   * Moves to the previous page, when the carousel is activated.
+   */
+  public async previousPage() {
+    this.currentPage =
+      this.currentPage - 1 < 0 ? this.numberOfPages - 1 : this.currentPage - 1;
+  }
+
+  /**
+   * Moves to the next page, when the carousel is activated.
+   */
+  public async nextPage() {
+    this.currentPage = (this.currentPage + 1) % this.numberOfPages;
+  }
+
+  public initialize() {
+    this.validateProps();
+    this.validateSlotID();
+    this.initRecommendations();
+    this.initSummary();
+    this.initProductTemplateProvider();
+    this.initProductListCommon();
+    this.createSelectChildProductListener();
+    createAppLoadedListener(this.bindings.store, (isAppLoaded) => {
+      this.isAppLoaded = isAppLoaded;
+    });
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    this.unsubscribeSummary && this.unsubscribeSummary();
+    this.removeEventListener(
+      'atomic/selectChildProduct',
+      this.selectChildProductCallback
+    );
+  }
+
+  constructor() {
+    super();
+    const contextRoot = new ContextRoot();
```

Comment:
This will no longer be needed when we merge the migrated atomic-commerce-recommendation-interface, correct?

If so, we should add a TODO to remove this.